flake.nix

```diff
@@ -49,25 +45,39 @@
         let
           pkgs = pkgsFor system;
 
-          rust = pkgs.rust-bin.fromRustupToolchainFile ./rust-toolchain;
+          rust = fenix.packages.${system}.fromToolchainFile {
+            file = ./rust-toolchain;
+            sha256 = "sha256-6PfBjfCI9DaNRyGigEmuUP2pcamWsWGc4g7SNEHqD2c=";
+          };
 
-          naersk-lib = naersk.lib."${system}".override {
+          buildRustPackage = (pkgs.makeRustPlatform {
             cargo = rust;
             rustc = rust;
-          };
+          }).buildRustPackage;
         in
         rec {
           # `nix build`
-          packages.${name} = naersk-lib.buildPackage {
+          packages.${name} = buildRustPackage {
+            inherit name;
             pname = name;
-            root = ./.;
+            src = nixpkgs.lib.cleanSource ./.;
+            cargoLock.lockFile = ./Cargo.lock;
+
+            preBuildPhases = [ "codeStyleConformanceCheck" ];
+
+            codeStyleConformanceCheck = ''
+            export PATH="${rust}/bin:$PATH";
+            # rustfmt
+            cargo fmt -- --check
```

Comment:
now fails fast before building anything :tada:

This slipped into this refactoring since I neede to figure out how to re-implement `cargoTestCommands` which now has become very tellingly `codeStyleConformanceCheck`.